eth-bytecode-db/eth-bytecode-db/src/verification/db.rs

```diff
@@ -131,21 +111,116 @@ async fn insert_source_files(
                 .context("insert into \"files\"")?,
             }
         };
+        result.push(file);
+    }
+
+    Ok(result)
+}
+
+async fn insert_source_details(
+    txn: &DatabaseTransaction,
+    source: types::Source,
+    file_models: &[files::Model],
+) -> Result<(sources::Model, bool), anyhow::Error> {
+    let abi = match source.abi {
+        None => None,
+        Some(abi) => serde_json::from_str(&abi).context("deserialize abi")?,
+    };
+
+    let file_ids = {
+        let mut file_ids = file_models.iter().map(|file| file.id).collect::<Vec<_>>();
+        file_ids.sort();
+        file_ids.dedup();
+        file_ids
+    };
+    // Results in `SELECT md5(array_agg(x)::text) FROM (VALUES (1), (2), (3)) t(x).
+    // Without t() `array_agg(x)::text` results in {(1),(2),(3)} instead of {1,2,3}.
+    let file_ids_hash_query = format!(
+        "SELECT md5(array_agg(x)::text)::uuid FROM (VALUES {}) t(x)",
+        file_ids
+            .iter()
+            .map(|i| format!("({i})"))
+            .collect::<Vec<_>>()
+            .join(",")
+    );
+    let file_ids_hash: sea_orm::prelude::Uuid = txn
+        .query_one(Statement::from_string(
+            DatabaseBackend::Postgres,
+            file_ids_hash_query,
+        ))
+        .await
+        .context("calculate hash of file ids")?
+        .ok_or(anyhow::anyhow!(
+            "selection of file ids resulted in empty result"
+        ))
+        .context("calculate hash of file ids")?
+        .try_get("", "md5")
+        .context("calculate hash of file ids")?;
+
+    let (source, inserted) = {
+        let compiler_settings: sea_orm::prelude::Json =
+            serde_json::from_str(&source.compiler_settings)
+                .context("deserialize compiler settings")?;
+
+        let db_source = sources::Entity::find()
+            .filter(sources::Column::CompilerVersion.eq(source.compiler_version.clone()))
+            .filter(sources::Column::CompilerSettings.eq(compiler_settings.clone()))
```

Comment:
Didn't you calculate hashes specifically to remove this comparison? Or does it compare hashes?

Reply:
May be I got something mixed up

Reply:
The hash is calculated to enforce uniqueness property, because the index data can be of a limited size only. Here we use that data instead, as we can compare the whole data without any problem. I'm not sure about performance, but it is just much easier to do using `sea-orm` framework. In case of any performance issues, I believe we may update that filtering